components/ImprovedSearch/Pagination.tsx

```diff
@@ -0,0 +1,116 @@
+import { Icon } from '@equinor/eds-core-react'
+import { chevron_left, chevron_right, first_page, last_page } from '@equinor/eds-icons'
+import { useContext, useEffect, useRef } from 'react'
+import { usePagination, UsePaginationProps } from 'react-instantsearch-hooks-web'
+import styled from 'styled-components'
+import { usePrefersReducedMotion } from './ReducedMotion'
+import { PaginationContext } from './PaginationContext'
+import { PaginationItem } from './PaginationItem'
+
+
+const PaginationList = styled.ul`
+  list-style: none;
+  padding: 0;
+  margin: 0;
+  display: grid;
+  grid-gap: 8px;
+  grid-auto-columns: min-content;
+  grid-auto-flow: column;
+`
+
+export type PaginationProps = {
+  hitsPerPage?: number
+  inverted?: boolean
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+} & UsePaginationProps
+
+export const Pagination = ({ totalPages, padding, hitsPerPage = 5, inverted = false, ...rest }: PaginationProps) => {
+  const { refine, createURL, pages, currentRefinement, isFirstPage, isLastPage, nbPages, nbHits } = usePagination({
+    totalPages,
+    padding,
+  })
+
+  const { resultsRef } = useContext(PaginationContext)
+  const prevRefinement = useRef<number>(currentRefinement)
+  const prefersReducedMotion = usePrefersReducedMotion()
```

Comment:
I think we can leave this as is for now. But just a friendly reminder about this media query and the scroll to top-animation: Animations is not actually something we have in our code base, so we should see if it fits or not.